backend/server/user_api_test.go

```
@@ -0,0 +1,183 @@
+		}
+	})
+
+	t.Run("Profile - Success", func(t *testing.T) {
+		user := &model.User{
+			ID:    "test-user-id",
+			Email: "profile@example.com",
+			Name:  "Profile User",
+		}
+
+		req := httptest.NewRequest(http.MethodGet, "/profile", nil)
+		req = req.WithContext(context.WithValue(req.Context(), "user", user))
```

golangci-lint (1.64.8)

SA1029: should not use built-in type string as key for value; define your own type to avoid collisions

(staticcheck)